inmem_endpoint.go

```diff
@@ -0,0 +1,133 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+	"strings"
+)
+
+// MetricsSummary holds a roll-up of metrics info for a given interval
+type MetricsSummary struct {
+	Timestamp string
+	Gauges    []GaugeValue
+	Points    []PointValue
+	Counters  []SampledValue
+	Samples   []SampledValue
+}
+
+type GaugeValue struct {
+	Name   string
+	Value  float32
+	Labels map[string]string
+}
+
+type PointValue struct {
+	Name   string
+	Points []float32
+}
+
+type SampledValue struct {
+	Name string
+	*AggregateSample
+	Mean   float64
+	Stddev float64
+	Labels map[string]string
+}
+
+// DisplayMetrics returns a summary of the metrics from the most recent finished interval.
+func (i *InmemSink) DisplayMetrics(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	i.intervalLock.Lock()
+	defer i.intervalLock.Unlock()
+
+	var interval *IntervalMetrics
+	n := len(i.intervals)
+	switch {
+	case n == 0:
+		return nil, fmt.Errorf("no metric intervals have been initialized yet")
+	case n == 1:
+		interval = i.intervals[0]
+	default:
+		interval = i.intervals[n-2]
+	}
+
+	summary := MetricsSummary{
+		Timestamp: interval.Interval.String(),
+		Gauges:    []GaugeValue{},
```

Comment:
Are these not left `nil`-valued so they show up as empty arrays in the JSON output?